common/shell/commands/flash_shell.c

```diff
@@ -0,0 +1,111 @@
+#include "flash_shell.h"
+
+#include <devicetree.h>
+#include <device.h>
+#include <stdio.h>
+#include "libutil.h"
+
+/* 
+    Command FLASH
+*/
+void cmd_flash_re_init(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc != 2) {
+		shell_warn(shell, "Help: platform flash re_init <spi_device>");
+		return;
+	}
+
+	const struct device *flash_dev;
+	flash_dev = device_get_binding(argv[1]);
+
+	if (!flash_dev) {
+		shell_error(shell, "Can't find any binding device with label %s", argv[1]);
+	}
+
+	if (spi_nor_re_init(flash_dev)) {
+		shell_error(shell, "%s re-init failed!", argv[1]);
+		return;
+	}
+
+	shell_print(shell, "%s re-init success!", argv[1]);
+	return;
+}
+
+void cmd_flash_sfdp_read(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc != 2 && argc != 3 && argc != 4) {
+		shell_warn(
+			shell,
+			"Help: platform flash sfdp_read <spi_device> <offset(optional, default 0)> <read_bytes(optional, default 256)>");
+		return;
+	}
+
+	int read_bytes = SFDP_BUFF_SIZE;
+	int offset = 0;
+	if (argc == 4) {
+		offset = strtol(argv[2], NULL, 16);
+		read_bytes = strtol(argv[3], NULL, 10);
+	} else if (argc == 3) {
+		offset = strtol(argv[2], NULL, 10);
+	}
+
+	const struct device *flash_dev;
+	flash_dev = device_get_binding(argv[1]);
+
+	if (!flash_dev) {
+		shell_error(shell, "Can't find any binding device with label %s", argv[1]);
+	}
+
+	if (!device_is_ready(flash_dev)) {
+		shell_error(shell, "%s: device not ready", flash_dev->name);
+		return;
+	}
+
+	uint8_t *raw = malloc(read_bytes * sizeof(uint8_t));
```

Comment:
We should probably define some max number we will allow.

Reply:
got it, thanks.